boa/util/lrudict.py

```diff
@@ -10,7 +10,7 @@ def __getitem__(self, k):
         return val
 
     def __setitem__(self, k, val):
-        if len(self) == self.n:
+        if len(self) == self.n and k not in self:
```

Comment:
i don't think this change is correct, it breaks the LRU semantics of the lru cache

Reply:
when `k` is in `self` we don't need to remove anything, otherwise we end up with a cache with less items than requested.
for example this test results in a dict with only 9 items:
https://github.com/vyperlang/titanoboa/blob/0a8640f049f8b4703f09b998eaef36be01be0316/tests/unitary/utils/test_lrudict.py#L16